tests/test_route.py

```diff
@@ -524,49 +490,43 @@ def test_RouteAndNexthopInDifferentVrf(self, dvs, testlog):
         time.sleep(1)
 
         # check application database
-        tbl = swsscommon.Table(self.pdb, "ROUTE_TABLE:Vrf_1")
-        route_entries = tbl.getKeys()
+        route_entries = self.pdb.get_keys("ROUTE_TABLE:Vrf_1")
         assert "20.0.1.2" in route_entries
 
-        tbl = swsscommon.Table(self.pdb, "ROUTE_TABLE:Vrf_2")
-        route_entries = tbl.getKeys()
+        route_entries = self.pdb.get_keys("ROUTE_TABLE:Vrf_2")
         assert "10.0.0.2" in route_entries
 
         # check ASIC neighbor interface database
-        tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP")
-        nexthop_entries = tbl.getKeys()
+        nexthop_entries = self.adb.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP")
         for key in nexthop_entries:
-            (status, fvs) = tbl.get(key)
+            fvs = self.adb.get_entry("ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP", key)
+            status = bool(fvs)
             assert status == True
-            for fv in fvs:
-                if fv[0] == "SAI_NEXT_HOP_ATTR_IP" and fv[1] == "20.0.1.2":
-                    nexthop2_found = True
-                    nexthop2_oid = key
-                if fv[0] == "SAI_NEXT_HOP_ATTR_IP" and fv[1] == "10.0.0.2":
-                    nexthop1_found = True
-                    nexthop1_oid = key
+            if fvs["SAI_NEXT_HOP_ATTR_IP"] == "20.0.1.2":
+                nexthop2_found = True
+                nexthop2_oid = key
+            if fvs["SAI_NEXT_HOP_ATTR_IP"] == "10.0.0.2":
+                nexthop1_found = True
+                nexthop1_oid = key
 
         assert nexthop1_found == True and nexthop2_found == True
```

Comment:
This should be wrapped in a `wait_until`, see example here: https://github.com/Azure/sonic-swss/blob/3052fb5f25fea0ba2f60ad17c294e6da8b9eb70a/tests/conftest.py#L351

Reply:
Wrapped these into check_route_entries_nexthop() function.